### ISSUE TITLE
Slide and Report Deliverables

### ISSUE THREAD

**Issue by alt113** [traffic-control-fyp-aub/ns3-gym#29]
Go thru the final slides and report templates on Moodle to check what would be needed in terms of deliverables, charts, and screen shots from NGS

Linked to #40 .

**Comment by alt113** [comment]
@AhmadAbouAdla1 any updates on this?

**Comment by AhmadAbouAdla1** [comment]
Working on it right now. Will provide a checklist hopefully tonight.

**Comment by AhmadAbouAdla1** [comment]
[Fall Semester FYP presentation.pptx](https://github.com/traffic-control-fyp-aub/ns3-gym/files/4427480/Fall.Semester.FYP.presentation.pptx)
Attached is the merged final presentation. 
I have the following remarks:
-Mention the relevant protocols at the end of the literature review
-Added Proposed solution methodology after the final design decision 
-Design alternatives and decision for ??? (didn’t understand the purpose of this slide)
-Design Iterations: Actions that resulted in changes in the overall design, constraint, or specification choices
-Implementation: Include photos + screenshots + videos
-Testing and Analysis: Details of designed experiment and testing plan. Testing results must be detailed and analyzed to show that they satisfy the requirements and specifications
-Conclusion: Analyze overall design and outcomes + Highlight advantages and limitations of design + Address significant broader impact

Thus, we need screenshots of any additional scenario (2 lane, merge, ...) and results of simulations that help confirm that the algorithm actually mitigates stop and go waves. If you feel that there is any additional info, videos or images that are crucial to be added, also send.
Moreover, we also need graphs that show the limitations of the system (ie bottlenecks).
As for the iterations, if you could also lead me through the process of what happened, changed along the way, and what major decisions affected the entire design, that would be great. 
In the mean time, I will be doing the same process for the report. I will use your answers for both.
(@jKh98 @alt113 )

**Comment by alt113** [comment]
@AhmadAbouAdla1 sounds fair, I think we have a lot of what you mentioned already. Whenever you can go through it one more time and maybe specify in more details and exactly what is needed (e.g speeds with no agent vs with agent, etc...).

**Comment by AhmadAbouAdla1** [comment]
Deal, will provide a more comprehensive and specific list very soon

**Comment by AhmadAbouAdla1** [comment]
This is a preliminary list for the things that we should add to the final presentation:
-Design Iterations: 
	Include any choices that show how we reached the decisions we are at now

-Implementation:
	Recommended to include photos, screenshots, videos
	Screenshots of environment during each phase:
		-Rectangle Scenario
		-Single Lane Highway Scenario
		-2 lane scenario (if we’ll implement it)
	
-Testing and Analysis:
	Details of Designed experiment 
    	Screenshots of testing results  (Graphs to show performance of system during each phase with each ML algorithm used (most important ones from those you showed me before) )+ Show that these results satisfy the requirements and specifications

-Conclusion:
	Advantage of design
	Limitations of design (ie throughput)

**Comment by alt113** [comment]
@AhmadAbouAdla1 okay that was definitely clearer. Could you now check to see what we already have and don't have and then edit the most recent comment you made on this thread to reflect that (a simple check next to what we do already have for example). 

***Note**: Try checking in the following folder `rl_fyp/ns3_logs/` there are three directories currently in there and each has a lot of data so we should be able to make use of what is in there.

**Comment by AhmadAbouAdla1** [comment]
On it

**Comment by AhmadAbouAdla1** [comment]
Alright so from the requirements, the ones we have are checked:

- [x] Square Scenario (can we include vehicle and RSU nodes?)

- [x] Single Lane Highway Scenario (can we include vehicle and RSU nodes?)

- [ ] Two-Lane Scenario (can we include vehicle and RSU nodes?)

- [x] System Performance under Square Scenario with PPO2

- [x] System Performance under Square Scenario with SAC

- [x] System Performance under Square Scenario with TD3

- [x] System Performance under Highway Scenario with PPO2

- [x] System Performance under Highway Scenario with SAC

- [x] System Performance under Highway Scenario with TD3

- [ ] ~~System Performance under 2-lane Highway Scenario with PPO2~~
`We showed for the 1-lane that it does not even converge in training so its kind of irrelevant to run PPO2 in the denser 2-lane scenario.`

- [ ] System Performance under 2-lane Highway Scenario with SAC

- [ ] System Performance under 2-lane Highway Scenario with TD3

- [ ] ~~System Performance under very high throttling (bottleneck/throughput) to demonstrate limitations of design~~
`We are always running the system in a congested traffic state so this isn't necessary.`

- [x] ~~Plot of reward (how is this integrated into the context?)~~
`The reward plot is for us only, to view the performance of the agent and if it is stabilizing at a good value, which is how we deduced that PPO2 failed in the 1-lane highway.`

- ~~We then need to show that PPO2 is not able to generalize to a realistic flow on a single lane highway (does not converge) while SAC and TD3 are promising but the system speed average is very low, knowing that the throughput is very high. Moreover, SAC is very realistic for simulations but not for real-life driving scenarios and events, which TD3 is. I am confused with how to demonstrate this without being too wordy in the presentation (or if we should include it at all).~~
`I've prepared a report on findings that suggest that in fact the throttled speeds are necessary given the dense traffic conditions on the highway. I'll send it on the FYP group now.`

- As for the initial training, we will first demonstrate that we initially trained the agent continuously on the square scenario, which is good since it is a closed scenario, then after the agent learns how to mitigate stop and go waves with forceful stopping every 30 seconds, we evaluate is the agent has learned correctly (how would we demonstrate that the agent learned correctly?). After that, we reloaded the agent into the highway scenario and continue training like before. (Also, how do we demonstrate that the desired results are achieved?)
`Answered in report that I will send now.`

- It is very important to showcase our way of thinking throughout the project, and to demonstrate the choices and design iterations that we took. Thus, my first take on this is that we started with square scenario(closed...) then onto highway(...) and we tried each algorithm respectively and moved on to the other when our needs are not met. If I am missing anything, please let me know.
`Didn't quite get this point, is it a question or are you telling us what you had in mind?`

**Comment by alt113** [comment]
@AhmadAbouAdla1 Noted, but some parts are already done and decided on so check your last comment on this thread now that I've updated it.

**Comment by alt113** [comment]
@AhmadAbouAdla1 I've said in the main thread that this is linked to #40 . i'm going to close this and shift focus to that issue instead to keep track of what's left in the final couple of weeks before deliverables are due.
